src/sendGCM.js

```diff
@@ -106,7 +108,8 @@ const sendGCM = (regIds, data, settings) => {
       data: data.custom,
     };
   }
-
+  custom.id = custom.id || data.id;
```

Comment:
is it needed in both places?

Reply:
yes. i tested this at work the last few days.
Maybe we should rename the data notifications to data notifications instead of "phone gap" mode

Reply:
You can have data notifications without what we call "phonegap mode".
```
data: opts.phonegap === true ? Object.assign(custom, notification) : custom, // See https://github.com/phonegap/phonegap-plugin-push/blob/master/docs/PAYLOAD.md#android-behaviour
```